scipy/stats/tests/test_stats.py

```diff
@@ -2469,6 +2471,17 @@ def test_onesample(self):
             assert_raises(ValueError, stats.ttest_1samp, x, 5.0,
                           nan_policy='foobar')
 
+    def test_1samp_alternative(self):
+        assert_raises(ValueError, stats.ttest_1samp, self.X1, 0,
+                      alternative="error")
+
+        t, p = stats.ttest_1samp(self.X1, 1, alternative="less")
+        assert_array_almost_equal(p, self.P1_1_l)
```

Comment:
Here and elsewhere, use `assert_allclose` instead of `assert_array_almost_equal` in the tests that you add.  While we have a lot of uses of `assert_array_almost_equal` in the existing code, in new code we should be using `assert_allclose`.  (I think we're all still working on adapting to this preferred API--I've used `assert_array_almost_equal` in recent PRs myself.)